### DISCUSSION TITLE
mqtt qos2

### DISCUSSION THREAD

**Discussion by zzzlll111** [cesanta/mongoose#2836] [Ask questions, get answers] [answered]
Hello please can you write a more detailed mqtt public qos2 quality routine?

**Comment by scaprile** [comment, marked as answer]
I'm sorry, current docs is as far as we will go with this. Any proficient developer should be able to understand that and write application code
https://github.com/cesanta/mongoose/issues/2689
Please see our [documentation](https://mongoose.ws/documentation/),  and follow the guidelines in our [tutorials](https://mongoose.ws/documentation/#tutorials-and-examples).
https://mongoose.ws/documentation/tutorials/mqtt/mqtt-client/#sending-qos-2-messages
https://github.com/cesanta/mongoose/blob/22b52f831bfd103b70a9b27209a0d2dd7df2fe1e/test/unit_test.c#L606-L635
